lib/src/sql/subquery.rs

```diff
@@ -222,6 +225,29 @@ impl Display for Subquery {
 	}
 }
 
+impl Serialize for Subquery {
+	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+	where
+		S: serde::Serializer,
+	{
+		if is_internal_serialization() {
+			match self {
+				Self::Value(v) => serializer.serialize_newtype_variant(TOKEN, 0, "Value", v),
+				Self::Ifelse(v) => serializer.serialize_newtype_variant(TOKEN, 1, "Ifelse", v),
+				Self::Output(v) => serializer.serialize_newtype_variant(TOKEN, 2, "Output", v),
+				Self::Select(v) => serializer.serialize_newtype_variant(TOKEN, 3, "Select", v),
+				Self::Create(v) => serializer.serialize_newtype_variant(TOKEN, 4, "Create", v),
+				Self::Update(v) => serializer.serialize_newtype_variant(TOKEN, 5, "Update", v),
+				Self::Delete(v) => serializer.serialize_newtype_variant(TOKEN, 6, "Delete", v),
+				Self::Relate(v) => serializer.serialize_newtype_variant(TOKEN, 7, "Relate", v),
+				Self::Insert(v) => serializer.serialize_newtype_variant(TOKEN, 8, "Insert", v),
+			}
+		} else {
+			serializer.serialize_some(&self.to_string())
```

Comment:
We should never get to serialising this type as a basic JSON/CBOR output, as we will always have called `compute` on this value by this time, and converted it to a primary type (Null, False, True, Number, Strand, Duration, Datetime, Uuid, Array, Object, Geometry). Therefore maybe a `serialize_none()` is better here?